token/src/components/token/erc721/erc721_metadata.cairo

```diff
@@ -0,0 +1,108 @@
+use starknet::ContractAddress;
+
+///
+/// Model
+///
+
+#[derive(Model, Copy, Drop, Serde)]
+struct ERC721MetaModel {
+    #[key]
+    token: ContractAddress,
+    name: felt252,
+    symbol: felt252,
+    base_uri: felt252,
+}
+
+///
+/// Interface
+///
+
+#[starknet::interface]
+trait IERC721Metadata<TState> {
+    fn name(self: @TState) -> felt252;
+    fn symbol(self: @TState) -> felt252;
+    fn token_uri(self: @TState, token_id: u128) -> felt252;
```

Comment:
we should use`ByteArray` to handle strings (not sure if its yet supported in Models, but should be soon)